core/runtime/jsruntime.rs

```diff
@@ -901,6 +901,16 @@ impl JsRuntime {
               panic!("{} not present in the module map", specifier)
             })
         };
+        {
+          let module_map_rc = self.module_map.clone();
+          let module_map = module_map_rc.borrow();
+          let handle = module_map.handles.get(mod_id).unwrap().clone();
+          let mut scope = realm.handle_scope(self.v8_isolate());
+          let handle = v8::Local::new(&mut scope, handle);
+          if handle.get_status() == v8::ModuleStatus::Evaluated {
```

Comment:
Maybe I'm misunderstanding, but what if the new module is slightly different than the one found in the snapshot? Shouldn't we panic instead since this isn't supported? https://github.com/denoland/deno_core/pull/39

Reply:
Maybe we could add a debug assert to check that the content is the same. Then again it's not likely to happen with the `extension!()` declarations. And IMO it's conceptually similar to encountering an import of an already-evaluated module with the loader in a new state with new content for that module; it would be standard to ignore it in that case.

If things surrounding module evaluation were more idempotent like with V8 / ESM semantics, I've seen that it would simplify and limit the need for conditional compilation in certain uses.

Reply:
Someone could be loading stuff dynamically though and not realize this, then be super confused when their updated code wasn't updated because a past snapshot had the old out of date code. I think we should hard fail if someone provides the same module twice and it's more efficient to enforce that.